File: gwt-time/gwt-time/src/main/java/org/jresearch/threetenbp/gwt/time/client/SupportJs.java
package org.jresearch.threetenbp.gwt.time.client;

import elemental2.core.ArrayBuffer;
import elemental2.core.Uint8Array;
import jsinterop.annotations.JsPackage;
import jsinterop.annotations.JsType;

@JsType(isNative = true, namespace = JsPackage.GLOBAL, name = "window")
public class SupportJs {

  public static native float getTimestamp();

  public static native String getTimezone();

  public static native void sleep(int milliseconds);

  /**
   * @param style "short", "long"
   * @param locale language tag
   */
  public static native String displayTimeZone(
      boolean daylight, String timeZone, String style, String locale);

  /**
   * @param style "2 digit", "numeric", "narrow", "short", "long"
   * @param locale language tag
   */
  public static native String[] displayMonths(String style, boolean standalone, String locale);

  /**
   * @param style "narrow", "short", "long"
   * @param locale language tag
   */
  public static native String[] displayWeekdays(String style, boolean standalone, String locale);

  /**
   * @param style "narrow", "short", "long"
   * @param locale language tag
   */
  public static native String[] displayEras(String style, String locale);

  /**
   * @param style "narrow", "short", "long"
   * @param locale language tag
   */
  public static native String[] displayAmpm(String style, String locale);

  public static native DecimalProperty displayNumber(String locale);

  public static native String[] supportedLocalesOfDateTimeFormat(String[] locales);

  public static native String[] supportedLocalesOfNumberFormat(String[] locales);

  @JsType(isNative = true, namespace = JsPackage.GLOBAL, name = "Base64Binary")
  public static class Base64Binary {

    public static native ArrayBuffer decodeArrayBuffer(String base64);

    public static native Uint8Array decode(String base64);
  }
}

File: gwt-time/gwt-time/src/main/java/org/jresearch/threetenbp/gwt/time/client/SupportJs.js
/**
 * @externs
 */
 
/**
* @return {number}
*/
var getTimestamp = function() {};
 
/**
* @return {string}
*/
var getTimezone = function() {};
 
/**
* @param {number} milliseconds
*/
var sleep = function(milliseconds) {};
 
/**
* @return {string}
* @param {boolean} daylight
* @param {string} timeZone
* @param {string} style
* @param {string} locale
*/
var displayTimeZone = function(daylight, timeZone, style, locale) {};
 
/**
* @return {Array<string>}
* @param {string} style
* @param {boolean} standalone
* @param {string} locale
*/
var displayMonths = function(style, standalone, locale) {};
 
/**
* @return {Array<string>}
* @param {string} style
* @param {boolean} standalone
* @param {string} locale
*/
var displayWeekdays = function(style, standalone, locale) {};
 
/**
* @return {Array<string>}
* @param {string} style
* @param {string} locale
*/
var displayEras = function(style, locale) {};
 
/**
* @return {Array<string>}
* @param {string} style
* @param {string} locale
*/
var displayAmpm = function(style, locale) {};
 
/**
* @return {!DecimalProperty}
* @param {string} locale
*/
var displayNumber = function(locale) {};
 
/**
* @return {Array<string>}
* @param {Array<string>} locales
*/
var supportedLocalesOfDateTimeFormat = function(locales) {};
 
/**
* @return {Array<string>}
* @param {Array<string>} locales
*/
var supportedLocalesOfNumberFormat = function(locales) {};
 
/**
 * @record
 */
var DecimalProperty;
 
/** @type {string} */ DecimalProperty.prototype.zeroDigit;
/** @type {string} */ DecimalProperty.prototype.positiveSign;
/** @type {string} */ DecimalProperty.prototype.negativeSign;
/** @type {string} */ DecimalProperty.prototype.decimalSeparator;
 
var Base64Binary = function() {}
 
/**
 * @param {string} base64
 * @return {ArrayBuffer}
 */
Base64Binary.decodeArrayBuffer = function(base64) {};
 
/**
 * @param {string} base64
 * @return {Uint8Array}
 */
Base64Binary.decode = function(base64) {};